src/v2/manifest/mod.rs

```diff
@@ -162,15 +162,18 @@ impl Client {
             .build_reqwest(reqwest::async::Client::new().get(url.clone()))
             .headers(accept_headers)
             .send()
-            .map_err(|e| Error::from(format!("{}", e)))
+            .map_err(Error::from)
             .inspect(move |_| {
                 trace!("HEAD {:?}", url);
             })
-            .and_then(|r| {
+            .and_then(|r| -> std::result::Result<_, ::errors::Error> {
```

Comment:
Isnt't this return type the same as `::errors::Result<_>`?
(I'm not sure why this needs additional hinting, but I trust that you fought the compiler before ending up with this)

Reply:
This was part of debugging and leftover. Thanks for catching!